### PR TITLE
Backpressure fetcher.

### PR DESCRIPTION
**Pull request by goral09**
Await with setting the timeout after we're confident that the request had been passed to the kernel.

### CODE REVIEW

**Comment by marc-casperlabs** [comment]
Looks good to me!

File: node/src/components/fetcher.rs [hunk 1]
```diff
@@ -151,18 +151,14 @@ pub(crate) trait ItemFetcher<T: Item + 'static> {
         id: T::Id,
         peer: NodeId,
     ) -> Effects<Event<T>> {
+        let peer_timeout = self.peer_timeout();
```
Comment (marc-casperlabs) [inline node/src/components/fetcher.rs:154]: If I wanted to be pedantic, I could say we can move this inside the `Ok` branch. But we will never trigger the `Err` anway :P

**Review by marc-casperlabs — APPROVED** [review]

File: node/src/components/fetcher.rs (hunk 1 above)
Comment (goral09) [reply node/src/components/fetcher.rs:154]: I see, so you would do:
```rust
Ok(message) => {
  let peer_timout = ...
  async move { ... }
}
```
I can do that :+1:

Comment (marc-casperlabs) [reply node/src/components/fetcher.rs:154]: Not

```rust
 Ok(message) => async move {
                let peer_timeout = self.peer_timeout();
                effect_builder.send_message(peer, message).await;

                effect_builder.set_timeout(peer_timeout).await
```

but

```rust
 Ok(message) => {
        let peer_timeout = self.peer_timeout();
        async move {
                effect_builder.send_message(peer, message).await;

                effect_builder.set_timeout(peer_timeout).await
```

Comment (marc-casperlabs) [reply node/src/components/fetcher.rs:154]: But let's keep as is, easier on the eyes :)

**Pull request merged**